tasks/CommandInjection.cpp

```diff
@@ -0,0 +1,83 @@
+/* Generated from orogen/lib/orogen/templates/tasks/Task.cpp */
+
+#include "CommandInjection.hpp"
+
+using namespace auv_control;
+
+CommandInjection::CommandInjection(std::string const& name)
+    : CommandInjectionBase(name)
+{
+}
+
+CommandInjection::CommandInjection(std::string const& name, RTT::ExecutionEngine* engine)
+    : CommandInjectionBase(name, engine)
+{
+}
+
+CommandInjection::~CommandInjection()
+{
+}
+
+bool CommandInjection::calcOutput()
+{
+    base::LinearAngular6DCommand output_command = merged_command;
+
+    if(_cmd_injection.connected())
+    {
+        base::LinearAngular6DCommand cmd_injection;
+        if(_cmd_injection.readNewest(cmd_injection) == RTT::NewData)
```

Comment:
You have no guarantee that there *is* a data read here since you don't test with RTT::OldData. Moreover, don't use readNewest. It's really unnecessary, and is problematic when running a component against log streams.

Moreover deeply, imbricated ifs() are really bad form. Avoid them as much as you can.

Just replace the two imbricated ifs by

```
RTT::FlowState state = _cmd_injection.read(cmd_injection);
if (state == RTT::NoData)
   return;
else if (state == RTT::OldData)
{
   if ((base::Time::now() - newest_injection_sample).toSeconds() <= _cmd_injection_timeout.value())
       return;
}
else if (state == RTT::NewData)
   newest_injection_sample = base::Time::now();

// Now do the injection and write
```